### PR TITLE
MantidWorkspaceProvider unit tests

### LINKED ISSUE #156: Unit tests for MantidWorkspaceProvider
The MantidWorkspaceProvider class currently has no unit tests.

### PR DESCRIPTION
**Pull request by MatthewBowles**
Added unit tests for MantidWorkspaceProvider class.

Fixes #156

### CODE REVIEW

**Comment by MatthewBowles** [comment]
@martyngigg Here's a start. I'm not sure if I can/should test some of the other stuff (e.g. the other routes through `_processLoadedWSLimits`) since they seem tied up in the mantid workspace code. Thoughts?

**Comment by martyngigg** [comment]
This is a good start. I do think tests for `get_limits` (so ultimately  `processLoadedWSLimits`) are required as it's quite complicated and thus something could feasibly go wrong with a calculation and it could go unnoticed.

**Comment by MatthewBowles** [comment]
@martyngigg I've added some tests for `get_limits`.

File: mslice/models/workspacemanager/mantid_workspace_provider.py [hunk 1]
```diff
@@ -60,9 +74,6 @@ def _processLoadedWSLimits(self, workspace):
             else:
                 self._limits[ws_name] = None
                 return
-        except AttributeError: # Wrong workspace type (e.g. cut)
```
Comment (martyngigg) [inline mslice/models/workspacemanager/mantid_workspace_provider.py]: I think we still need these lines? If we don't have them then the nice work from #147 breaks.

File: mslice/models/workspacemanager/mantid_workspace_provider.py [hunk 2]
```diff
@@ -60,9 +74,6 @@ def _processLoadedWSLimits(self, workspace):
             else:
                 self._limits[ws_name] = None
```
Comment (martyngigg) [inline mslice/models/workspacemanager/mantid_workspace_provider.py]: I know you didn't change this but could we use `self._limits[ws_name] = {}` here instead for consistency with other places here?

**Comment by MatthewBowles** [comment]
@martyngigg OK. I've tried to tidy up `_processLoadedWSLimits` while I was there since it was way too long.

**Pull request merged**